ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/utils.py

```diff
@@ -0,0 +1,287 @@
+
+        return None
+
+    def _get_adapter_storm_control(self, adapter: dict) -> dict | None:
+        """
+        Return storm_control for one adapter
+        """
+        if self._platform_settings_feature_support_interface_storm_control:
+            return get(adapter, "storm_control")
+
+        return None
+
+    def _get_adapter_evpn_ethernet_segment_cfg(
+        self,
+        adapter: dict,
+        short_esi: str,
+        node_index: int,
+        connected_endpoint: dict,
+        default_df_algo: str = None,
+        default_redundancy: str = None,
+    ) -> dict | None:
+        """
+        Return evpn_ethernet_segment_cfg for one adapter
+        """
+        if short_esi is None:
+            return None
+
+        adapter_ethernet_segment: dict = adapter.get("ethernet_segment", {})
+        evpn_ethernet_segment = {
+            "identifier": generate_esi(short_esi, self._evpn_short_esi_prefix),
+            "redundancy": adapter_ethernet_segment.get("redundancy", default_redundancy),
+            "route_target": generate_route_target(short_esi),
+        }
+        if (designated_forwarder_algorithm := adapter_ethernet_segment.get("designated_forwarder_algorithm", default_df_algo)) is None:
+            return evpn_ethernet_segment
+
+        if designated_forwarder_algorithm == "modulus":
+            evpn_ethernet_segment["designated_forwarder_election"] = {"algorithm": "modulus"}
+
+        elif designated_forwarder_algorithm == "auto":
+            auto_preferences = range((len(adapter["switches"]) - 1) * 100, -1, -100)
+            evpn_ethernet_segment["designated_forwarder_election"] = {
+                "algorithm": "preference",
+                "preference_value": auto_preferences[node_index],
+                "dont_preempt": adapter_ethernet_segment.get("dont_preempt"),
+            }
+
+        elif designated_forwarder_algorithm == "preference":
+            # TODO: Add check for length of designated_forwarder_preferences
+            designated_forwarder_preferences = get(
+                adapter_ethernet_segment,
+                "designated_forwarder_preferences",
+                required=True,
+                org_key=f"ethernet_segment.designated_forwarder_preferences for the connected_endpoint {connected_endpoint['name']}",
+            )
+            evpn_ethernet_segment["designated_forwarder_election"] = {
+                "algorithm": "preference",
+                "preference_value": designated_forwarder_preferences[node_index],
+                "dont_preempt": adapter_ethernet_segment.get("dont_preempt"),
+            }
+
+        return evpn_ethernet_segment
+
+    def _get_adapter_link_tracking_groups(self, adapter: dict) -> dict | None:
+        """
+        Return link_tracking_groups for one adapter
+        """
+        if not (get(adapter, "link_tracking.enabled") is True and self._link_tracking_groups is not None):
```

Comment:
written this way it allows to not do the `get` if the cached property is None anyway
```suggestion
        if self._link_tracking_groups is None or get(adapter, "link_tracking.enabled") is not True:
```

Reply:
done